cpp_ext/TorchValues.cpp

```diff
@@ -131,16 +132,27 @@ void PyAnyTorchListValue::bindDerived(ClassTy &c) {
   py::implicitly_convertible<py::tuple, PyAnyTorchListValue>();
 }
 
-#define DEFINE_LIST_BASE_CONCRETE_VALUE(TORCHTYPE)                             \
+#define DEFINE_LIST_BASE_CONCRETE_VALUE(TORCHTYPE, SCALARTYPE)                 \
   void PyAnyTorchListOf##TORCHTYPE##Value::bindDerived(ClassTy &c) {           \
     c.def(py::init<py::list>(), py::arg("value"));                             \
     c.def(py::init<py::tuple>(), py::arg("value"));                            \
+    c.def(                                                                     \
+        "__getitem__",                                                         \
+        [](PyAnyTorchListOf##TORCHTYPE##Value & self,                          \
+           const PyTorch_IntValue &idx) -> PyTorch_##SCALARTYPE##Value {       \
+          MlirType containedType = torchMlirTorchListTypeGetContainedType(     \
+              mlirValueGetType(self.get()));                                   \
+          return PyGlobals::get()                                              \
+              .lookupOperationClass("torch.aten.__getitem__.t")                \
+              .value()(py::cast(containedType), self, idx)                     \
+              .cast<PyTorch_##SCALARTYPE##Value>();                            \
+        });                                                                    \
```

Comment:
does this work? these only differ by return type and overloading based on return [doesn't work in cpp](https://stackoverflow.com/a/9568890).

Reply:
These also differ based on the input list type, however, so the signature should be different right? These are also members of each List class so they shouldn't conflict. Additionally, it does pass the unit tests I included. Does that address what you mean?

Reply:
oh my bad you're right they differ by the pair (torchtype, scalartype). yea all good.